### DISCUSSION TITLE
OSR与NCW切换问题

### DISCUSSION THREAD

**Discussion by njuzhou** [CefView/QCefView#146] [Q&A] [answered]
请问切换OSR和NCW模式，是需要重新构建源码吗，还是可以在代码中通过addCommandLineSwitch来设置呢

**Comment by catcheroftime** [comment, marked as answer]
目前我看说明QCefView是通过重新构建源码
以generate-win-x86.bat脚本为例，通过添加 -DUSE_OSR=OFF，重新构建
注意清空之前编译的缓存

**Reply by njuzhou** [reply]
嗯嗯，我也看是的，只是通过指引中的下载qcefview和core代码后，进行generate-win-x86-64和cmake --build后，总是会出现一些问题。
你在嵌入后，有遇到渲染性能的问题吗，我这里是因为嵌入的网页数据量比较大，导致和谷歌浏览器加速速度差异比较大，所以想看看怎样提高性能，看说明用ncw可以提高，所以在尝试

**Reply by catcheroftime** [reply]
ncw目前测试基本可以达到和浏览器差不多的效果

**Reply by njuzhou** [reply]
你也是构建了一个ncw模式的库吗？能否发一下，真的遇到难题了

**Reply by njuzhou** [reply]
知道这样的请求可能不太好，但确实项目遇到了困难，如果方便的话还希望能够分享哈

**Comment by tishion** [comment]
OSR和NCW只能在生项目时候选择。
性能问题是因为使用的QT的Software渲染引擎，如果是对性能要求的应用，建议使用NCW模式。NCW在linux平台的兼容性有一些问题，不太适用于生产环境。

**Reply by njuzhou** [reply]
我这边主要基于windows桌面系统，那应该问题不大，感谢作者，我来研究下如何生成NCW版本库

**Reply by njuzhou** [reply]
另外请问下，在windows生产环境下用这个模式，会有什么潜在问题吗

**Reply by tishion** [reply]
没什么问题

**Reply by njuzhou** [reply]
好的，感谢！
